addons/spreadsheet/static/src/list/plugins/list_core_plugin.js

```diff
@@ -151,7 +152,49 @@ export default class ListCorePlugin extends CorePlugin {
             case "REMOVE_GLOBAL_FILTER":
                 this._onFilterDeletion(cmd.id);
                 break;
+
+            case "START":
+                for (const sheetId of this.getters.getSheetIds()) {
+                    const cells = this.getters.getCells(sheetId);
+                    for (const cell of Object.values(cells)) {
+                        if (cell.isFormula()) {
+                            this._addListPositionToDataSource(cell.content);
+                        }
+                    }
+                }
+                break;
+            case "UPDATE_CELL":
+                if (cmd.content) {
+                    this._addListPositionToDataSource(cmd.content);
+                }
+                break;
+        }
+    }
+
+    /**
+     * Extract the position of the records asked in the given formula and
+     * increase the max position of the corresponding data source.
+     *
+     * @param {string} content Odoo list formula
+     */
+    _addListPositionToDataSource(content) {
+        if (getNumberOfListFormulas(content) !== 1) {
+            return;
+        }
+        const { functionName, args } = getFirstListFunction(content);
```

Comment:
actually, there's no reason to limit this to the first list function. Requires a bit more work though